packages/mobile/src/screens/search-screen-v2/searchState.ts

```diff
@@ -53,3 +58,34 @@ export const useSearchFilter = <F extends SearchFilter>(filterKey: F) => {
 
   return [filter, setFilter, clearFilter] as const
 }
+
+type SearchResultsApiType = ReturnType<typeof useGetSearchResultsApi>
+
+type SearchResultsType<C extends SearchCategory> = {
+  status: SearchResultsApiType['status']
+  data: C extends 'all'
+    ? SearchResultsApiType['data']
+    : SearchResultsApiType['data'][Exclude<C, 'all'>]
```

Comment:
Fancy

Reply:
yeah kinda a lot... potentially not worth it